src/main/java/gregtech/common/entities/GTBoatEntity.java

```diff
@@ -0,0 +1,171 @@
+                this.entityDropItem(MetaItems.TREATED_WOOD_BOAT.getStackForm(), 0);
+                break;
+        }
+    }
+
+    // override to change boat drop
+    @Override
+    public boolean attackEntityFrom(DamageSource source, float amount) {
+        if (this.isEntityInvulnerable(source)) {
+            return false;
+        }
+        if (!this.world.isRemote && !this.isDead) {
+            if (source instanceof EntityDamageSourceIndirect && source.getTrueSource() != null && this.isPassenger(source.getTrueSource())) {
+                return false;
+            }
+            this.setForwardDirection(-this.getForwardDirection());
+            this.setTimeSinceHit(10);
+            this.setDamageTaken(this.getDamageTaken() + amount * 10);
+            this.markVelocityChanged();
+            boolean flag = source.getTrueSource() instanceof EntityPlayer && ((EntityPlayer) source.getTrueSource()).capabilities.isCreativeMode;
+
+            if (flag || this.getDamageTaken() > 40) {
+                if (!flag && this.world.getGameRules().getBoolean("doEntityDrops")) {
+                    dropBoatItems(false);
+                }
+
+                this.setDead();
+            }
+
+        }
+        return true;
+    }
+
+    // override to change boat drop
+    @Override
+    protected void updateFallState(double y, boolean onGroundIn, IBlockState state, BlockPos pos) {
+        this.lastYd = this.motionY;
+
+        if (!this.isRiding()) {
+            if (onGroundIn) {
+                if (this.fallDistance > 3) {
+                    if (this.status != EntityBoat.Status.ON_LAND) {
+                        this.fallDistance = 0;
+                        return;
+                    }
+
+                    this.fall(this.fallDistance, 1);
+
+                    if (!this.world.isRemote && !this.isDead) {
+                        this.setDead();
+                        if (this.world.getGameRules().getBoolean("doEntityDrops")) {
+                            dropBoatItems(true);
+                        }
+                    }
+                }
+
+                this.fallDistance = 0;
+            } else if (this.world.getBlockState((new BlockPos(this)).down()).getMaterial() != Material.WATER && y < 0) {
+                this.fallDistance = (float) ((double) this.fallDistance - y);
+            }
+        }
+    }
+
+    @Override
+    public ItemStack getPickedResult(RayTraceResult ray) {
+        switch (getGTBoatType()) {
+            case RUBBER_WOOD_BOAT:
+                return MetaItems.RUBBER_WOOD_BOAT.getStackForm();
+            case TREATED_WOOD_BOAT:
+                return MetaItems.TREATED_WOOD_BOAT.getStackForm();
+            default:
+                return ItemStack.EMPTY;
+        }
+    }
+
+    @Deprecated
+    @Override
+    public void setBoatType(Type type) {
+        super.setBoatType(type);
+    }
+
+    @Deprecated
+    @Override
+    public Type getBoatType() {
+        return super.getBoatType();
+    }
```

Comment:
These methods should have javadocs stating which methods to defer to.

Reply:
The "Type" referred here is for vanilla boat types which shouldn't be used on GTBoat. I'll write some javadocs explaining the deprecation.